### PR TITLE
Jetpack Checkout: Set post purchase destination to wp-admin recommendations page.

### PR DESCRIPTION
**Pull request by elliottprogrammer**
This PR updates the post-purchase destination (for non-atomic Jetpack purchases) to the site's wp-admin Jetpack dashboard recommendations assistant.

Previously these purchases were being redirected to the Calypso blue product checklist (guided tour).

Related to: 1164141197617539-as-1200403218057318

#### Changes proposed in this Pull Request

* Renamed the flag `redirectCloudCheckoutToWpAdmin` to `redirectCheckoutToWpAdmin`
* And now use this flag for both Cloud and non-Cloud (Calypso) checkout to determine the post-checkout destination.
* As of this PR, `redirectCheckoutToWpAdmin` is `true` and set to `/wp-admin/admin.php?page=jetpack#/recommendations` for all non-atomic Jetpack purchases originating from **_both_** Calypso checkout and Cloud checkout. (however Cloud checkout is not currently in use at this time anyway, but if/when it is enabled someday, it is also set to the Recommendations assistant page.)

#### Testing instructions

1. Checkout this PR & run `yarn start`
2. Create a JN site and connect it (Or use any self-hosted Jetpack site already connected to your account)
3. Select the site and go to the /plans page: `http://calypso.localhost:3000/plans/:site`
4. Select and purchase any product or plan.
5. Verify that after purchase completed, you are redirected to the site's recommendation assistant.
    ( `/wp-admin/admin.php?page=jetpack#/recommendations` )

**Verify all unit tests pass:**
Run: `yarn test-client client/my-sites/checkout/composite-checkout/test/composite-checkout-thank-you.js`

### CODE REVIEW

**Review by rcanepa — APPROVED** [review]
LGTM! I tested this starting from WordPress.com's Plans page and Jetpack's Pricing page and it worked as expected in both cases. 

I made sure that it didn't break the new logged-out checkout flow. 

🚢🚢

**Comment by rcoll** [comment]
This seems fine to me also, but there are a couple failing e2e tests that need to be fixed before shipping.

**Review by sirbrillig — APPROVED** [review]
No objections from payments. Thanks for adding tests!

**Comment by sirbrillig** [comment]
This had to be reverted in https://github.com/Automattic/wp-calypso/pull/53596 because it breaks (due to encoding of the `#` in the URL) our PayPal express flow. We are working on a backend solution to allow the return URL to include a `#` but in the meantime reverting this will allow PayPal purchases to work again for Jetpack sites.

**Comment by elliottprogrammer** [comment]
Ok, thanks for reverting. Let me know when a solution is in place, or if there is anything I can do to help. 👍

**Comment by sirbrillig** [comment]
The underlying issue should be fixed by D62649-code. Once that is merged, this PR should be able to be recreated safely.

**Comment by nbloomf** [comment]
The backend fix has been merged.

**Comment by sirbrillig** [comment]
@elliottprogrammer Can we leave it to your team to recreate this PR? Sorry for the inconvenience this caused. (If you'd prefer that we took care of it, let me know.)

**Comment by elliottprogrammer** [comment]
Yes, no problem at all! We'll take care of it. Thanks for catching this and for implementing a fix so quickly! 👍

**Pull request merged**
